superset/commands/dataset/create.py

```diff
@@ -85,10 +88,12 @@ def validate(self) -> None:
             exceptions.append(TableNotFoundValidationError(table_name))
 
         if sql:
+            processor = jinja_context.get_template_processor(database=database)
+            rendered_sql = processor.process_template(sql)
```

Comment:
The additional challenge is that—depending on the macro—an actual query may need to be executed in order to process the template, i.e., the `[hive|presto|trino].latest_partition` Jinja macro actually executes a query in order to resolve the macro. This results in somewhat of a chicken-and-egg scenario, i.e., one has to _try_ to execute a query (on behalf of the user) in order to resolve whether the user has access to it.

Furthermore if the user doesn't have access to the tables which are used to resolve the `[hive|presto|trino].latest_partition` Jinja macro then line #92 will raise an `SupersetDBAPIProgrammingError` (or similar).

I'm still not sure what the ideal solution is here. Without doubt this is non-trivial.